core/src/main/java/com/linecorp/armeria/client/Http2ResponseDecoder.java

```diff
@@ -245,11 +245,17 @@ public int onDataRead(
         }
 
         final long maxContentLength = res.maxContentLength();
-        if (maxContentLength > 0 && res.writtenBytes() > maxContentLength - dataLength) {
-            res.close(ContentTooLargeException.get());
-            throw connectionError(INTERNAL_ERROR,
-                                  "content length too large: %d + %d > %d (stream: %d)",
-                                  res.writtenBytes(), dataLength, maxContentLength, streamId);
+        final long transferredLength = res.writtenBytes();
+        if (maxContentLength > 0 && transferredLength > maxContentLength - dataLength) {
+            res.close(ContentTooLargeException.builder()
+                                              .limit(maxContentLength)
+                                              .delta(dataLength)
+                                              .transferred(transferredLength)
+                                              .build());
+            throw connectionError(
+                    INTERNAL_ERROR,
+                    "content length too large: transferred(%d) + total(%d) > maximum(%d) (stream: %d)",
```

Comment:
```suggestion
                    "content length too large: transferred(%d) + delta(%d) > maximum(%d) (stream: %d)",
```

Reply:
total -> delta, maximum -> limit :-)

Reply:
> total -> delta, maximum -> limit :-)

🤦🏻‍♂️ @trustin @ikhoon thank you all for reviewing 🙇🏻